sdk/textanalytics/azure-ai-textanalytics/tests/test_detect_language.py

```diff
@@ -0,0 +1,569 @@
+
+        def callback(resp):
+            self.assertIn("azsdk-python-azure-ai-textanalytics/{} Python/{} ({})".format(
+                VERSION, platform.python_version(), platform.platform()),
+                resp.http_request.headers["User-Agent"]
+            )
+
+        docs = [{"id": "1", "text": "I will go to the park."},
+                {"id": "2", "text": "I did not like the hotel we stayed it."},
+                {"id": "3", "text": "The restaurant had really good food."}]
+
+        response = text_analytics.detect_language(docs, raw_response_hook=callback)
+
+    @GlobalTextAnalyticsAccountPreparer()
+    def test_document_attribute_error_no_result_attribute(self, resource_group, location, text_analytics_account, text_analytics_account_key):
+        text_analytics = TextAnalyticsClient(text_analytics_account, TextAnalyticsApiKeyCredential(text_analytics_account_key))
+
+        docs = [{"id": "1", "text": ""}]
+        response = text_analytics.detect_language(docs)
+
+        # Attributes on DocumentError
+        self.assertTrue(response[0].is_error)
+        self.assertEqual(response[0].id, "1")
+        self.assertIsNotNone(response[0].error)
+
+        # Result attribute not on DocumentError, custom error message
+        try:
+            primary_language = response[0].primary_language
+        except AttributeError as custom_error:
+            self.assertEqual(
+                custom_error.args[0],
+                '\'DocumentError\' object has no attribute \'primary_language\'. '
+                'The service was unable to process this document:\nDocument Id: 1\nError: '
+                'invalidDocument - Document text is empty.\n'
+            )
+
+    @GlobalTextAnalyticsAccountPreparer()
+    def test_document_attribute_error_nonexistent_attribute(self, resource_group, location, text_analytics_account, text_analytics_account_key):
+        text_analytics = TextAnalyticsClient(text_analytics_account, TextAnalyticsApiKeyCredential(text_analytics_account_key))
+
+        docs = [{"id": "1", "text": ""}]
+        response = text_analytics.detect_language(docs)
+
+        # Attribute not found on DocumentError or result obj, default behavior/message
+        try:
+            primary_language = response[0].attribute_not_on_result_or_error
+        except AttributeError as default_behavior:
+            self.assertEqual(
+                default_behavior.args[0],
+                '\'DocumentError\' object has no attribute \'attribute_not_on_result_or_error\''
+            )
+
+    @GlobalTextAnalyticsAccountPreparer()
+    def test_text_analytics_error_bad_model_version(self, resource_group, location, text_analytics_account, text_analytics_account_key):
+        text_analytics = TextAnalyticsClient(text_analytics_account, TextAnalyticsApiKeyCredential(text_analytics_account_key))
+        text = ""
```

Comment:
Remove the code that prepares the large doc for the series of document error tests (applies to async file as well)